tests/basic/uncontrolled_randomness.rs

```diff
@@ -0,0 +1,231 @@
+use rand::{rngs::StdRng, SeedableRng};
+use shuttle::rand::Rng;
+use shuttle::scheduler::{DfsScheduler, Scheduler};
+use shuttle::sync::{Arc, Mutex};
+use shuttle::{check_uncontrolled_randomness, thread};
+use test_log::test;
+
+fn check_uncontrolled_randomness_custom_scheduler_and_config<F, S>(f: F, scheduler: S)
+where
+    F: Fn() + Send + Sync + 'static,
+    S: Scheduler + 'static,
+{
+    use shuttle::scheduler::UncontrolledRandomnessCheckScheduler;
+    use shuttle::Config;
+
+    let config = Config::default();
+
+    let scheduler = UncontrolledRandomnessCheckScheduler::new(scheduler);
+
+    let runner = shuttle::Runner::new(scheduler, config);
+    runner.run(f);
+}
+
+fn randomly_acquire_lock<F: (Fn() -> Box<dyn rand::RngCore>) + Send + Sync>(thread_rng: &'static F) {
```

Comment:
```suggestion
fn randomly_acquire_lock<R: rand::RngCore, F: (Fn() -> R) + Send + Sync>(thread_rng: &'static F) {
```
and then remove all the `Box`es at the callsites